### ISSUE TITLE
[Bug report] Duplicate node modules getting bundled

### ISSUE THREAD

**Issue by jrson83** [Zhengqbbb/cz-git#107]
### ✅ Checklist

- [X] I am using the **latest version**
- [X] I have searched [FAQ](https://cz-git.qbb.sh/faq/)

### 📄 System Info

```Shell
System:
  OS: Windows 10 10.0.22621
  CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
  Memory: 17.64 GB / 31.93 GB
Binaries:
  Node: 18.14.0 - ~\AppData\Local\Volta\tools\image\node\18.14.0\node.EXE
  Yarn: 1.22.18 - ~\AppData\Local\Volta\tools\image\yarn\1.22.18\bin\yarn.CMD
  npm: 9.3.1 - ~\AppData\Local\Volta\tools\image\node\18.14.0\npm.CMD
npmPackages:
  cz-git: workspace:* => 1.6.0
  czg: workspace:* => 1.6.0
```

#### Concise Description

There are duplicate node modules in the monorepo, which get bundled twice on build.

#### Expected behavior

Each module should be bundled once.

#### Steps to reproduce

1. Clone and install the monorepo.

```bash
git clone https://github.com/Zhengqbbb/cz-git.git
cd cz-git
pnpm install
```

Tsup can generate a `metafile` for [bundle analysis](https://tsup.egoist.dev/#metafile) by adding `--metafile` as flag.

2. Open `packages/cli/package.json` and add `--metafile` to the build script.

```json
"build": "pnpm clean && tsup --metafile"
```

3. Run `pnpm build`.
4. Goto [Bundle Buddy](https://www.bundle-buddy.com), choose `esbuild`, click the button Upload assets `ESBUILD.JSON` and upload the metafile from `packages/cli/lib/metafile-cjs.json`.
5. Scroll down and click `GO TO ANALYSIS`.
6. You should see the duplicate modules in health check section.

#### Screenshots or Screen Recording

![2023-03-17_13h17_31](https://user-images.githubusercontent.com/45284565/225902140-eb645d57-6ec5-4eec-835d-d4d1e5a8281c.png)

#### Possible solution

Use [pnpm overrides](https://pnpm.io/package_json#pnpmoverrides) in package.json to get rid of the duplicate modules:

```json
"pnpm": {
  "overrides": {
    "chalk": "^5.2.0",
    "color-convert": "^2.0.1",
    "supports-color": "^9.3.1"
  },
}
```

I have tested this and the bundle-size is about 1kb less after and the duplicate modules are gone:

![2023-03-17_13h30_42](https://user-images.githubusercontent.com/45284565/225905311-16148cd7-5521-44c1-ad9a-8ed5fa5dc7e4.png)

I was hoping this helps, but after build all tests fail, cause [chalk](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) and [supports-color](https://github.com/chalk/supports-color/releases/tag/v9.0.0) dropped cjs support.

I will test again with the last versions with cjs support. If I find another solution I will let you know.

**UPDATE:**

I tested again and with the following versions, the tests run successful (added to root package.json):

```json
"pnpm": {
  "overrides": {
    "chalk": "4.1.2",
    "color-convert": "2.01",
    "supports-color": "8.1.1"
  },
}
```

**Comment by jrson83** [comment]
I will create the PR, but before releasing we should test in detail if it doesn't break anything.

**Comment by Zhengqbbb** [comment]
That's for sure
 I will close If the 1KB brings uncertainty

**Comment by Zhengqbbb** [comment]
I have released the next version 1.6.2-beta.1 
Feel free to use it and let me know if there are any problems with it!

**Comment by jrson83** [comment]
Awesome, thank you!
